src/controller/submissions.controller.ts

```diff
@@ -0,0 +1,68 @@
+import { Submission } from 'devu-shared-modules'
+import { Request, Response, NextFunction } from 'express'
+
+import SubmissionService from '../services/submission.service'
+
+import { GenericResponse, NotFound } from '../utils/apiResponse.utils'
+
+import { serialize } from '../utils/serializer/submissions.serializer'
+
+export async function get(req: Request, res: Response, next: NextFunction) {
+  try {
+    const submissions = await SubmissionService.list()
+    const response = submissions.map(serialize)
+
+    res.status(200).json(response)
+  } catch (err) {
+    next(err)
+  }
+}
+
+export async function detail(req: Request, res: Response, next: NextFunction) {
+  try {
+    const id = parseInt(req.params.id)
+    const submission = await SubmissionService.retrieve(id)
+
+    if (!submission) return res.status(404).json(NotFound)
+
+    const response = serialize(submission)
+
+    res.status(200).json(response)
+  } catch (err) {
+    next(err)
+  }
+}
+
+export async function post(req: Request, res: Response, next: NextFunction) {
+  try {
+    // If this is hit, developer messed up above in the chain by not checking the user for auth
+    if (!req.currentUser?.userId) return res.status(400).json(new GenericResponse('Request requires auth'))
```

Comment:
Should we be doing this since it's only to catch developer errors? If yes, we should add it to every method in every controller which would be excessive. Or is this check only here because we're reading the userId in this function?

Reply:
If we're going down this road, it should return a 401 code

Reply:
Honestly I'd prefer not to have this line of code entirely. It's only here to satiate typescript.

Because `currentUser` can be undefined (aka not passing auth/ not being behind auth middleware) ts will yell about us index into current user. So we add this to say "hey if current user isn't here, and we're expecting it to be". 

Realistically, this should be a 500 because it means we screwed up in development. But I don't feel good about returning a 500 intentionally.